azurerm/internal/services/analysisservices/analysis_services_server_resource.go

```diff
@@ -245,13 +245,35 @@ func resourceArmAnalysisServicesServerUpdate(d *schema.ResourceData, meta interf
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server creation.")
+	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server update.")
 
 	id, err := parse.AnalysisServicesServerID(d.Id())
 	if err != nil {
 		return err
 	}
 
+	serverResp, err := client.GetDetails(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Analysis Services Server %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	if serverResp.State != analysisservices.StateSucceeded && serverResp.State != analysisservices.StatePaused {
+		return fmt.Errorf("Error updating Analysis Services Server %q (Resource Group %q): State must be either Succeeded or Paused but got %q", id.Name, id.ResourceGroup, serverResp.State)
+	}
+
+	isPaused := serverResp.State == analysisservices.StatePaused
+
+	if isPaused {
+		resumeFuture, err := client.Resume(ctx, id.ResourceGroup, id.Name)
```

Comment:
Should we add a note to the docs that the resource will briefly unsuspend the server to make updates?

Reply:
Added a note for this in the docs.